### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by codepantha**
Changes introduced via this PR: 👇

- [x] Use Capybara to write integration tests for each view in your project.

### Login page:

- [x] I can see the username and password inputs and the "Submit" button.
- [x] When I click the submit button without filling in the username and the password, I get a detailed error.
- [x] When I click the submit button after filling in the username and the password with incorrect data, I get a detailed error.
- [x] When I click the submit button after filling in the username and the password with correct data, I am redirected to the root page.

### User index page:

- [x] I can see the username of all other users.
- [x] I can see the profile picture for each user.
- [x] I can see the number of posts each user has written.
- [x] When I click on a user, I am redirected to that user's show page.

### user show page:

- [x] I can see the user's profile picture.
- [x] I can see the user's username.
- [x] I can see the number of posts the user has written.
- [x] I can see the user's bio.
- [x] I can see the user's first 3 posts.
- [x] I can see a button that lets me view all of a user's posts.
- [x] When I click a user's post, it redirects me to that post's show page.
- [x] When I click to see all posts, it redirects me to the user's post's index page.

### User post index page:

- [x] I can see the user's profile picture.
- [x] I can see the user's username.
- [x] I can see the number of posts the user has written.
- [x] I can see a post's title.
- [x] I can see some of the post's body.
- [x] I can see the first comments on a post.
- [x] I can see how many comments a post has.
- [x] I can see how many likes a post has.
- [x] I can see a section for pagination if there are more posts than fit on the view.
- [x] When I click on a post, it redirects me to that post's show page.

### Post show page:

- [x] I can see the post's title.
- [x] I can see who wrote the post.
- [x] I can see how many comments it has.
- [x] I can see how many likes it has.
- [x] I can see the post body.
- [x] I can see the username of each commenter.
- [x] I can see the comment each commentor left.

## 🏃 To run the tests for this milestone:
`rspec spec/views/*_spec.rb`

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -2,15 +2,12 @@
...
-      bio: 'badass dev',
-      posts_counter: 0
-    )
+    User.create(name: 'Prom', bio: 'This is my life',
+                photo: 'https://avatars.githubusercontent.com/u/46839548?v=4',
+                email: 'prom@test.com', password: '123456')
   end
 
-  before { subject.save }
+  # before { subject.save }
```
Comment (arslanbisharat) [inline spec/models/user_spec.rb]: Please remove commented code from the project

File: spec/views/login_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# rubocop:disable Metrics/BlockLength
```
Comment (arslanbisharat) [inline spec/views/login_spec.rb]: Please don't disable linter rules as this makes code clean.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: spec/models/user_spec.rb (hunk 1 above)
Comment (codepantha) [reply spec/models/user_spec.rb]: Thank you for pointing this out @arslanbisharat. Fixed.

File: spec/views/login_spec.rb (hunk 2 above)
Comment (codepantha) [reply spec/views/login_spec.rb]: @arslanbisharat thank you for your review 🙏. Fixed

**Review by descholar-ceo — APPROVED** [review]
Hi @codepantherr , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**